File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/OnTheMove.java
package net.cyclestreets.liveride;

import net.cyclestreets.CycleStreetsPreferences;
import net.cyclestreets.routing.Journey;

final class OnTheMove extends MovingState
{
  OnTheMove(final LiveRideState previous) 
  {
    super(previous, CycleStreetsPreferences.nearingTurnDistance());
  } // OnTheMove
  
  @Override
  protected LiveRideState transitionState(final Journey journey)
  {
    return new NearingTurn(this, journey);
  } // transitionStatue
} // class OnTheMove
  

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/OnTheMove.kt
package net.cyclestreets.liveride

import net.cyclestreets.CycleStreetsPreferences
import net.cyclestreets.routing.Journey

internal class OnTheMove(previous: LiveRideState) :
        MovingState(previous, CycleStreetsPreferences.nearingTurnDistance()) {

    override fun transitionState(journey: Journey): LiveRideState {
        return NearingTurn(this, journey)
    }
}
